cg/apps/demultiplex/sample_sheet/index.py

```diff
@@ -127,34 +142,75 @@ def pad_index_two(index_string: str, reverse_complement: bool) -> str:
     return index_string + INDEX_TWO_PAD_SEQUENCE
 
 
-def adapt_indexes(
-    samples: List[FlowCellSample],
-    run_parameters: RunParameters,
+def get_hamming_distance_index_1(sequence_1: str, sequence_2: str) -> int:
+    """Get the hamming distance between two index 1 sequences.
+    In the case that one sequence is longer than the other, the distance is calculated between
+    the shortest sequence and the first segment of equal length of the longest sequence."""
+    limit: int = min(len(sequence_1), len(sequence_2))
+    return get_hamming_distance(str_1=sequence_1[:limit], str_2=sequence_2[:limit])
+
+
+def get_hamming_distance_index_2(sequence_1: str, sequence_2: str) -> int:
+    """Get the hamming distance between two index 2 sequences.
+    In the case that one sequence is longer than the other, the distance is calculated between
+    the shortest sequence and the last segment of equal length of the longest sequence."""
+    limit: int = min(len(sequence_1), len(sequence_2))
+    return get_hamming_distance(str_1=sequence_1[-limit:], str_2=sequence_2[-limit:])
+
+
+def adapt_barcode_mismatch_values_for_sample(
```

Comment:
What does adapt mean? Maybe the function name or doc string can be made more descriptive